cmd/pilot-agent/main.go

```diff
@@ -119,6 +125,30 @@ var (
 			proxyConfig.StatsdUdpAddress = statsdUDPAddress
 			proxyConfig.ProxyAdminPort = int32(proxyAdminPort)
 
+			var pilotSAN []string
+			switch controlPlaneAuthPolicy {
+			case int(proxyconfig.AuthenticationPolicy_NONE):
+				proxyConfig.ControlPlaneAuthPolicy = proxyconfig.AuthenticationPolicy_NONE
+			case int(proxyconfig.AuthenticationPolicy_MUTUAL_TLS):
+				var ns string
+				proxyConfig.ControlPlaneAuthPolicy = proxyconfig.AuthenticationPolicy_MUTUAL_TLS
+				if serviceregistry == platform.KubernetesRegistry {
+					partDiscoveryAddress := strings.Split(discoveryAddress, ":")
+					discoveryHostname := partDiscoveryAddress[0]
+					parts := strings.Split(discoveryHostname, ".")
+					if len(parts) == 1 {
+						// namespace of pilot is not part of discovery address use pod namespace
+						// e.g. istio-pilot:15003
+						ns = os.Getenv("POD_NAMESPACE")
+					} else {
+						// namespace is found in the discovery address
+						// e.g. istio-pilot.istio-system:15003
+						ns = parts[1]
+					}
+				}
+				pilotSAN = envoy.GetPilotSAN(pilotDomain, ns)
```

Comment:
silly question: isnt this common to all proxies?

Reply:
No. ingress and egress have istio-pilot:15003, whereas apps have istio-pilot.istio-system:15003
where namespace of istio components if istio-system.